### PR TITLE
Update Documentation: Add MacOS with Make instructions for Exercism CLI

### PR DESCRIPTION
**Pull request by cbullard-dev**
## Update Documentation: Add MacOS with Make instructions for Exercism CLI
This PR updates the documentation to improve the "MacOS with Make when using the Exercism CLI" section. The updated instructions clarify the process of configuring, testing, and submitting C++ exercises with the Exercism CLI on MacOS.

### Rationale:  
This update aligns the section with existing style and cadence in the documentation while providing additional clarity for users who may be unfamiliar with the nuances of the Exercism CLI setup. It aims to minimize confusion and ensure a smooth setup process on MacOS.

#### Impact:  
- Enhances the clarity of instructions for MacOS users.
- Provides explicit details to streamline the C++ exercise workflow with the Exercism CLI.

#### Testing:  
- Verified the commands and workflow on MacOS using `Unix Makefiles` with the Exercism CLI.
- Ensured documentation style and tone are consistent with existing content.

### CODE REVIEW

**Comment by IsaacG** [comment]
Relevant forum thread: https://forum.exercism.org/t/14261/

File: docs/TESTS.md [hunk 1]
```diff
@@ -108,10 +108,10 @@ Errors similar to `The CXX compiler identification is unknown` will likely be re
 [web-xcode-download]: https://apps.apple.com/us/app/xcode/id497799835?mt=12
 [cpp-installation-instructions]: https://exercism.org/docs/tracks/cpp/installation
 
-### Mac OS with Make _when using the exercise CLI_
-#### This example will specifically assist those who want to use the Exercism CLI app for configuring, testing and submitting their CPP examples.
+### MacOS with Make _when using the Exercism CLI_
+#### This example specifically assists those who want to use the Exercism CLI app for configuring, testing, and submitting their C++ exercises.
 
-The generator name for CMake is `Unix Makefiles`.
+The generator name for CMake is `Unix Makefiles`.
```
Comment (kotp) [inline docs/TESTS.md:114]:
```suggestion
The generator name for CMake is `Unix Makefiles`.
```

This should not have trailing white space for this line.

**Review by neenjaw — APPROVED** [review]

**Comment by vaeng** [comment]
Thank you for taking the time to update Mac instructions. There is another help file, that would benefit a Mac treatment, or do you think it is sufficient?

https://github.com/exercism/cpp/blob/main/exercises/shared/.docs/tests.md

**Comment by cbullard-dev** [comment]
I can take a look at it to see if I can run it based on that I can see if I can update the docs ✌🏼

**Comment by cbullard-dev** [comment]
> Thank you for taking the time to update Mac instructions. There is another help file, that would benefit a Mac treatment, or do you think it is sufficient?
> 
> 
> 
> https://github.com/exercism/cpp/blob/main/exercises/shared/.docs/tests.md 

I have read over the documentation here and this looks pretty universal and I don't think it would need Mac OS specific instruction as this has more to do with the TDD style of testing and how to iterate over the tests.

Let me know if you disagree ✌🏼

**Pull request merged**